metafix/src/main/java/org/metafacture/metafix/Metafix.java

```diff
@@ -247,6 +243,16 @@ public StreamReceiver getStreamReceiver() {
         return vars;
     }
 
+    public void putVar(final String key, final String value) {
+        try {
+            vars.put(key, value);
+        }
+        catch (final UnsupportedOperationException e) {
+            vars = new HashMap<>(vars);
```

Comment:
@fsteeg: WDYT? The alternative would be to keep a separate map and `putAll(newVars)` in `Metafix(Reader, Map<String, String>)`, like Metamorph does (via `AbstractMetamorphDomWalker`).

Reply:
Implemented above mentioned alternative in 887d6c6.